solver/src/settlement.rs

```diff
@@ -140,6 +140,13 @@ impl Settlement {
                 .get(&trade.order.buy_token)
                 .expect("Solution with trade but without price for buy token");
 
+            if trade.order.kind == OrderKind::Sell && buy_token_clearing_price.is_zero() {
+                return None;
+            }
+            if trade.order.kind == OrderKind::Buy && sell_token_clearing_price.is_zero() {
+                return None;
+            }
```

Comment:
Another way to write this could be

```suggestion
            if match trade.order.kind {
                OrderKind::Buy => &sell_token_clearing_price,
                OrderKind::Sell => &buy_token_clearing_price,
            }
            .is_zero()
            {
                return None;
            }

```

I like that because it uses match which I find clearer than if because it makes it obvious that the branches are exclusive and that all enum cases are handled. Either way is fine, just wanted to suggest it.

Reply:
Cool, thanks!